examples/c/ex_col_store.c

```diff
@@ -0,0 +1,519 @@
+    error_check(wiredtiger_open(home, NULL, "create,statistics=(fast)", &conn));
+
+    /* Establishing a session. */
+    error_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create a table with columns and colgroup. */
+    error_check(session->create(session, TABLE_NAME,
+      "key_format=r,value_format=" WT_UNCHECKED_STRING(
+        HHHHBBBB5S5S) ",columns=(id,hour,pressure,loc_lat,"
+                      "loc_long,temp,humidity,"
+                      "wind,feels_like_temp,day,country),colgroups=(day_time,temperature,"
+                      "humidity_pressure,"
+                      "wind,feels_like_temp,location)"));
+
+    /* Create the colgroups */
+    error_check(session->create(session, "colgroup:weather:day_time", "columns=(day,hour)"));
+    error_check(session->create(session, "colgroup:weather:temperature", "columns=(temp)"));
+    error_check(session->create(
+      session, "colgroup:weather:humidity_pressure", "columns=(humidity,pressure)"));
+    error_check(session->create(session, "colgroup:weather:wind", "columns=(wind)"));
+    error_check(
+      session->create(session, "colgroup:weather:feels_like_temp", "columns=(feels_like_temp)"));
+    error_check(
+      session->create(session, "colgroup:weather:location", "columns=(loc_lat,loc_long,country)"));
+
+    /* Generating random data to populate the weather table. */
+    generate_data(weather_data);
+
+    /* Open a cursor on the table to insert the data. */
+    error_check(session->open_cursor(session, TABLE_NAME, NULL, "append", &cursor));
+    for (int i = 0; i < NUM_ENTRIES; i++) {
+        cursor->set_value(cursor, weather_data[i].hour, weather_data[i].pressure,
+          weather_data[i].loc_lat, weather_data[i].loc_long, weather_data[i].temp,
+          weather_data[i].humidity, weather_data[i].wind, weather_data[i].feels_like_temp,
+          weather_data[i].day, weather_data[i].country);
+        error_check(cursor->insert(cursor));
+    }
+    /* Close cursor. */
+    error_check(cursor->close(cursor));
+
+    /* Prints all the data in the database. */
+    print_all_columns(session);
+
+    print_temp_column(session);
+
+    /* Update the temperature from Celsius to Fahrenheit. */
+    update_celsius_to_fahrenheit(session);
+
+    print_temp_column(session);
+
+    /* Create indexes for searching */
+    error_check(session->create(session, "index:weather:hour", "columns=(hour)"));
+    error_check(session->create(session, "index:weather:country", "columns=(country)"));
+
+    /*
+     * Start and end points for time range for finding min/max temperature, in 24 hour format.
+     * Example uses 10am - 8pm but can change the values for desired start and end times.
+     */
+    starting_time = 1000;
+    ending_time = 2000;
+    min_temp_result = 0;
+    max_temp_result = 0;
+
+    /* If the min/max temperature is not found due to some error, there is no result to print. */
+    if (find_min_and_max_temp(
```

Comment:
We are promoting a return type to boolean, hence losing the actual error returned. There might be unexpected errors for which you want to crash the program instead? If there are none like that, we still explicitly compare to success to make it obvious, for instance:

```
if (find_min_and_max_temp(...) == 0) {
    ....
}
```

Reply:
Would it be useful to store the result of the function call in a variable and then compare it to zero? I'm not sure if there are certain errors where we could need to crash the program and since it is an example, I have not done a whole lot of error checking.